### DISCUSSION TITLE
Network observer issues

### DISCUSSION THREAD

**Discussion by NSExceptional** [Flipboard/FLEX#495] [General]
This thread will serve as a place to discuss issues related to the Network History screen since third party libraries such as AFNetworking are prone to breaking it.

At the time of writing, ***the network observer works on iOS 14***. However, it is known that FLEX cannot observe requests made from a `WKWebView`. New issues about this shortcoming will be closed and directed here.

The start of this thread marks the closing of #293 and #478.

### WKWebView

Here is list of functions we would probably need to hook to add `WKWebView` support, since it appears `WKWebView` passes `NSURLRequest` objects to WebKit directly:

- [ ] `WebKit::WebPageProxy::loadFile`
- [ ] `WebKit::WebPageProxy::loadRequest`
- [ ] `WebKit::WebPageProxy::loadData`

**Comment by Cgema** [comment]
Excuse me,(*￣︶￣)~
Is this problem solved?
About observe requests made from a WKWebView。

**Reply by NSExceptional** [reply]
No it is not ;(

As a workaround I can recommend the Charles iOS app or Charles Mac app

**Comment by natedegoey** [comment]
Hi! I'm using FLEX in my project and I want to invoke the Network History page directly from a button in my app rather than through the traditional toolbar and globals menu. Is this possible, and if so, what would be the code to accomplish this? Thanks!

**Reply by NSExceptional** [reply]
Hi! You should just be able to create an instance of `FLEXNetworkMITMViewController` and present it however you wish. Off the top of my head I'm not sure if there's a public way to programmatically enable the network observer (it is off by default) but you can manually open flex and turn it on, and you only need to do that once.

Is there anything else I can help you with? :)

(Do let me know if you have any issues using `FLEXNetworkMITMViewController` in your project)

**Reply by natedegoey** [reply]
Thank you for the response! I've tried to use `FLEXNetworkMITMViewController` but it doesn't look like I can create an instance of that class.
![Screen Shot 2021-08-09 at 10 33 52 AM](https://user-images.githubusercontent.com/60538326/128723785-0c1da358-c7a9-42ce-9654-2bbf0b14cca1.png)
